v2/runners/coild_server.go

```diff
@@ -265,6 +266,17 @@ func (s *coildServer) Add(ctx context.Context, args *cnirpc.CNIArgs) (*cnirpc.Ad
 func (s *coildServer) Del(ctx context.Context, args *cnirpc.CNIArgs) (*empty.Empty, error) {
 	logger := ctxzap.Extract(ctx)
 
+	duration := 30 * time.Second
+	deadline, ok := ctx.Deadline()
+	if ok {
+		d2 := time.Until(deadline)
+		if d2 < duration && d2 > time.Second {
+			duration = d2 - time.Second
+		}
+	}
+	logger.Sugar().Infow("waiting before destroying pod network", "duration", duration.String())
+	time.Sleep(duration)
```

Comment:
If d2 is equal to the duration (or larger than the duration), this consumes all of the remaining deadline time.
If so, will calling s.nodeIPAM.Free fail always?

Another concern, is it enough to leave 1 second for the rest of the processing time?

Reply:
> If d2 is equal to the duration (or larger than the duration), this consumes all of the remaining deadline time.
> If so, will calling s.nodeIPAM.Free fail always?

Not necessarily, because `Del` is expected to be completed within 30 seconds usually.

> Another concern, is it enough to leave 1 second for the rest of the processing time?

Yes, again, usually.

Reply:
Ah, I misunderstood your first point.  Yes, it should wait shorter than `deadline`.

Reply:
Pushed a fix.

Reply:
I confirmed fixing my concern.
I am not familiar with CNI, so I hand off the entire review.